kubectl-cloudflow/config/config.go

```diff
@@ -217,25 +228,51 @@ func validateConfig(config *Config, applicationSpec cfapp.CloudflowApplicationSp
 			}
 		}
 	}
-	runtimesConfig := hoconConf.GetConfig(cloudflowRuntimesPrefix)
+	runtimesConfig := hoconConf.GetConfig(cloudflowRuntimesPath)
 	if runtimesConfig != nil && runtimesConfig.Root().IsObject() {
 		for runtime := range runtimesConfig.Root().GetObject().Items() {
 			runtimeConfig := runtimesConfig.GetConfig(runtime)
 			if runtimeConfig != nil && runtimeConfig.Root().IsObject() {
 				if runtimeConfig.GetConfig(configKey) == nil &&
 					runtimeConfig.GetConfig(kubernetesKey) == nil {
-					return fmt.Errorf("runtime config %s.%s does not contain '%s' or '%s' config sections", cloudflowRuntimesPrefix, runtime, configKey, kubernetesKey)
+					return fmt.Errorf("runtime config %s.%s does not contain '%s' or '%s' config sections", cloudflowRuntimesPath, runtime, configKey, kubernetesKey)
 				}
 				for runtimeConfigKey := range runtimeConfig.Root().GetObject().Items() {
 					if !(runtimeConfigKey == configKey || runtimeConfigKey == kubernetesKey) {
-						return fmt.Errorf("streamlet config %s.%s contains unknown section '%s'", cloudflowRuntimesPrefix, runtime, runtimeConfigKey)
+						return fmt.Errorf("streamlet config %s.%s contains unknown section '%s'", cloudflowRuntimesPath, runtime, runtimeConfigKey)
+					}
+				}
+			}
+		}
+	}
+	unknownTopics := []string{}
+	topicsConfig := hoconConf.GetConfig(cloudflowTopicsPath)
+	if topicsConfig != nil && topicsConfig.Root().IsObject() {
+		for topicID := range topicsConfig.Root().GetObject().Items() {
+			foundTopic := false
+			for _, deployment := range applicationSpec.Deployments {
+				for _, topic := range deployment.PortMappings {
+					if topic.ID == topicID {
+						foundTopic = true
+						break
 					}
 				}
 			}
+			if !foundTopic {
+				unknownTopics = append(unknownTopics, topicID)
+			}
 		}
 	}
-	if streamletsConfig == nil && runtimesConfig == nil {
-		return fmt.Errorf("Missing %s or %s config sections", cloudflowStreamletsPrefix, cloudflowRuntimesPrefix)
+	if len(unknownTopics) == 1 {
+		return fmt.Errorf("Unknown topic found in configuration file: %s", strings.Join(unknownTopics, ", "))
+	}
+
+	if len(unknownTopics) > 1 {
```

Comment:
is the plural "topic[s]" the only reason to duplicate this check? I would only keep one. If the plural form is really important, check the size to include the _s_ (or not).
```
unknownTopicLen := len(unknownTopics)
if unknownTopicLen >= 1 {
  if unknownTopicLen == 1 {
    pluralForm := ""  
  } else {
    pluralForm := "s"
  }
    return fmt.Errorf("Unknown topic%s found in configuration file: %s", pluralForm, strings.Join(unknownTopics, ", "))
  }
```